### PR TITLE
Add docs for PR (feedback)

### LINKED ISSUE #1063: Append guidelines with how PR feedback should be addressed
Pull request guidelines should contain something like "rule of a thumb" which rule how PR feedback should be addressed or how merge conflicts should be resolved.

As example consider #972. Commits like `Merge branch 'master' into Issue587` (aaa5434)
clutter history and should be avoided. `git rebase` should be used for updating branch with fix.
Other commits in this PR could be squashed.

As another example consider PR  #1003. It's initial commit of the library so it's OK to address feedback in a separate commit. However in this PR  feedback addressed twice and the second could be amended to the first.

Conclusion:
1. Merge conflicts should be resolved through rebasing.
2. PR feedback should be in a separate commit if it contains important changes. Changes like formatting or renaming variable introduced in PR are not important and could be addressed through amending/rebasing.

### PR DESCRIPTION
**Pull request by hughbe**
Let's hope this PR meets the guidelines

Fixes #1063

### CODE REVIEW

**Comment by hughbe** [comment]
Test Innerloop CentOS7.1 Debug Build and Test

**Comment by hughbe** [comment]
Innerloop CentOS7.1 Debug Build and Test

File: Documentation/project-docs/contributing.md [hunk 1]
```diff
@@ -16,3 +16,14 @@ API Changes
...
+Pull Requests
+-------------
+
+* All code changes should be submitted via pull requests (PRs) rather than direct commit access, and then merged after a peer review by one or more Microsoft employees.
+* PRs should have descriptive names (i.e. not "Fix #1234")
+* Within a PR, refer to any issues, and include [keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) that automatically close issues when the PR is merged, and tag any users to review or look at the change.
+* Try to make sure each commit can produce a successful build (and preferably passes tests).
+* If there is a merge conflict that needs fixing, prefer using `git rebase dotnet/master` instead of a merge commit.
+* PR feedback should be done in seperate commits, and should not be rebased or squashed unless necessary.
+* PRs should not be squashed before they are merged. If necessary, squashing should be handled by the merger using the ["Confirm squash and merge"](https://github.com/blog/2141-squash-your-commits) feature.
```
Comment (stephentoub) [inline Documentation/project-docs/contributing.md]: These all look good to me.  Could you rephrase them with the DO and DO NOT beginnings as in the other sections?

Comment (hughbe) [inline Documentation/project-docs/contributing.md]: Understood. Done

**Comment by stephentoub** [comment]
Thanks, Hugh.  I'll make some tweaks, but looks good.
